crates/matrix-sdk-ui/src/timeline/event_item/mod.rs

```diff
@@ -66,6 +69,11 @@ pub struct EventTimelineItem {
     pub(super) content: TimelineItemContent,
     /// The kind of event timeline item, local or remote.
     pub(super) kind: EventTimelineItemKind,
+    /// Whether or not the event belongs to an encrypted room.
+    ///
+    /// When `None` it is unknown if the room is encrypted and the item won't
+    /// return a ShieldState.
+    pub(super) is_room_encrypted: Option<bool>,
```

Comment:
Why is this an optional? The constructor takes a `bool`, when will this be `None`?

Reply:
When creating an item further in the `from_latest_event` constructor(?) it is set to `None` to avoid requesting the encryption state from the room list.

Reply:
Yeah it's the latter, it results in no shield info being available (which is fine at least for Element as we don't show it in the room list).

I would imagine this will need to change for Invisible Crypto but I'd consider that a conversation for another day?

Reply:
I certainly hope that we're not showing shields on rooms ever, as that is computationally not feasible to do fast, i.e. when the verification state of your own identity changes then the verification state and thus shield state of all identities changes.

This conversation leads me to believe that the latest event, i.e. the event we're using in the room list should be of a different type from what we use in the timeline, i.e. the room list shouldn't use `EventTimelineItem` because we're starting to diverge on what kind of data we need.

Could you add two things on the constructor `from_latest_event()`:
1.  Some docs that we will never produce a correct shield state if this constructor is used
2. A `TODO: ` item explaining that, due to a), we should move this functionality to another type, so we don't have such a footgun which might turn into a security issue.

Reply:
Fair, done in https://github.com/matrix-org/matrix-rust-sdk/pull/3783/commits/de502f42c61990e07ed450f84612789e3b8a934d

> I certainly hope that we're not showing shields on rooms ever

I wasn't specifically thinking shields here, but filtering out events that aren't trusted. If we ever start to do this in the timeline we would presumably also need to do the same in the room list. Although maybe by that point we'll be able to drive it from a local event cache or something :)